packages/in_app_purchase/lib/src/in_app_purchase/app_store_connection.dart

```diff
@@ -44,20 +44,21 @@ class AppStoreConnection implements InAppPurchaseConnection {
   Future<bool> isAvailable() => SKPaymentQueueWrapper.canMakePayments();
 
   @override
-  void buyNonConsumable({@required PurchaseParam purchaseParam}) {
-    _skPaymentQueueWrapper.addPayment(SKPaymentWrapper(
+  Future<bool> buyNonConsumable({@required PurchaseParam purchaseParam}) async {
+    await _skPaymentQueueWrapper.addPayment(SKPaymentWrapper(
         productIdentifier: purchaseParam.productDetails.id,
         quantity: 1,
         applicationUsername: purchaseParam.applicationUserName,
         simulatesAskToBuyInSandbox: purchaseParam.sandboxTesting,
         requestData: null));
+    return true; // There's no error feedback from iOS here to return.
```

Comment:
May be worth catching any `PlatformException`s here and returning false in that case? I didn't add it because I thought the behavior was a little too special cased, but that may be better than always returning true.

Reply:
Yeah, there shouldn't be any exception to be caught here. If there is an exception here, that means the developer has to fix it.